projects/js-packages/partner-coupon/components/redeem-partner-coupon-post-connection/test/component.jsx

```diff
@@ -0,0 +1,198 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import RedeemPartnerCouponPostConnection from '../';
+import analytics from '@automattic/jetpack-analytics';
+import { getRedirectUrl } from '@automattic/jetpack-components';
+
+const { location } = window;
+const partnerCoupon = {
+	coupon_code: 'TEST_TST_1234',
+	preset: 'TST',
+	partner: {
+		name: 'Company name',
+		prefix: 'TEST',
+		logo: {
+			src: 'IMAGE_SRC',
+			width: 150,
+			height: 100,
+		},
+	},
+	product: {
+		title: 'Awesome Product',
+		slug: 'awesome-product',
+		features: [ 'Feature 1', 'Feature 2', 'Feature 3' ],
+	},
+};
+const requiredProps = {
+	connectionStatus: { isRegistered: true },
+	partnerCoupon: partnerCoupon,
+	assetBaseUrl: 'PATH',
+	siteRawUrl: 'example.com',
+	tracksUserData: true,
+	analytics: analytics,
+};
+
+let locationAssignSpy;
+let recordEventStub;
+
+describe( 'RedeemPartnerCouponPostConnection', () => {
+	before( () => {
+		locationAssignSpy = sinon.spy();
+		recordEventStub = sinon.stub( analytics.tracks, 'recordEvent' );
+	} );
+
+	beforeEach( () => {
+		// Spy on location.assign, so we don't get breaking errors when
+		// we trigger click events on buttons/links.
+		delete window.location;
+		window.location = { assign: locationAssignSpy };
+	} );
+
+	afterEach( () => {
+		window.location = location;
+		locationAssignSpy.resetHistory();
+
+		recordEventStub.reset();
+	} );
+
+	after( () => {
+		recordEventStub.restore();
+	} );
+
+	it( 'is shown partner logo', () => {
+		render( <RedeemPartnerCouponPostConnection { ...requiredProps } /> );
+
+		const logo = screen.getByAltText(
+			'Logo of Company name who are offering a coupon in partnership with Jetpack'
+		);
+		expect( logo );
+		expect( logo.width ).equals( 150 );
+		expect( logo.height ).equals( 100 );
+	} );
+
+	it( 'do not try to show partner logo if we do not have any', () => {
+		const partnerCouponCopyWithoutLogo = partnerCoupon;
+		delete partnerCouponCopyWithoutLogo.partner.logo;
+
+		const props = {
+			...requiredProps,
+			partnerCoupon: partnerCouponCopyWithoutLogo,
+		};
+
+		const { container } = render( <RedeemPartnerCouponPostConnection { ...props } /> );
+
+		// We use querySelector because using typical screen.* selectors will give an
+		// error if the component doesn't exist; and we specifically want to ensure
+		// it doesn't exist while displaying the "Set up button".
+		expect(
+			container.querySelector(
+				'img[alt="Logo of Company name who are offering a coupon in partnership with Jetpack"]'
+			)
+		).to.not.exist;
```

Comment:
Other querySelector tests seem to use `.to.be.null;` - what would be the better option for all such tests?

Reply:
`to.not.exist` will return true for both `null` and `undefined` which IMO would both be acceptable values.

Technically `to.be.null` would be the "most strict comparison" since `querySelector` will return `null` if it cannot find anything, but what we care about is just that nothing is found, so I went with what I felt was most readable 😛 

I don't mind changing it if we want to be more strict with the comparison

Reply:
Sooo.. this might be just a liiiiiittle bit subjective since it's one of my previous PRs (😂) but we have similar conditions in the repo: 

https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/_inc/client/components/jetpack-product-card/test/component.js#L84